### PR TITLE
Improve plugin Code Coverage

### PR DESCRIPTION
**Pull request by prudhvigodithi**
### Description
This is the 1st PR as part of this [issue]( https://github.com/opensearch-project/job-scheduler/issues/612) to improve the repo code coverage gradually.
 
### Issues Resolved
Part of https://github.com/opensearch-project/job-scheduler/issues/612
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by prudhvigodithi** [comment]
FYI, I allowed the tests to run few times since yesterday and I get the following error.
```
{'detail': ErrorDetail(string='Rate limit reached. Please upload with the Codecov repository upload token to resolve issue. Expected available in 2831 seconds.', code='throttled')}
```
Let me add a Codecov token and update the test workflow to use the token for uploading the test report.
@bbarani @cwperks

**Comment by prudhvigodithi** [comment]
Works now https://app.codecov.io/gh/opensearch-project/job-scheduler/pull/616 and I see the percentage is now 37.27% (from 29.17%).
Screenshot 2024-04-29 at 11 02 41 AM
@cwperks @joshpalis can you please check?
@bbarani

File: src/test/java/org/opensearch/jobscheduler/JobSchedulerPluginTests.java [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    private Supplier<DiscoveryNodes> nodesInCluster;
+
+    @Before
+    public void setup() {
+        plugin = new JobSchedulerPlugin();
+        settings = Settings.builder()
+            .put(IndexMetadata.SETTING_VERSION_CREATED, Version.CURRENT)
+            .put(Environment.PATH_HOME_SETTING.getKey(), createTempDir().toString())
+            .build();
+        indexSettings = IndexSettingsModule.newIndexSettings(".opensearch-job-scheduler-job-details", settings);
```
Comment (cwperks) [inline src/test/java/org/opensearch/jobscheduler/JobSchedulerPluginTests.java:87]: Import this from JobDetailsService?

File: src/main/java/org/opensearch/jobscheduler/utils/JobDetailsService.java [hunk 2]
```diff
@@ -137,7 +137,7 @@ void updateIndexToJobProviders(String documentId, JobDetails jobDetails) {
      * @param documentId the unique Id for the job details
      * @param jobDetails the jobDetails to register
      */
-    void updateIndexToJobDetails(String documentId, JobDetails jobDetails) {
+    public void updateIndexToJobDetails(String documentId, JobDetails jobDetails) {
```
Comment (cwperks) [inline src/main/java/org/opensearch/jobscheduler/utils/JobDetailsService.java:140]: Is this change necessary?

File: src/main/java/org/opensearch/jobscheduler/JobSchedulerPlugin.java [hunk 3]
```diff
@@ -69,14 +69,13 @@ public class JobSchedulerPlugin extends Plugin implements ActionPlugin, Extensib
     public static final String JS_BASE_URI = "/_plugins/_job_scheduler";
 
     private static final Logger log = LogManager.getLogger(JobSchedulerPlugin.class);
+    JobSweeper sweeper;
```
Comment (cwperks) [inline src/main/java/org/opensearch/jobscheduler/JobSchedulerPlugin.java:72]: Are the changes to the access modifiers necessary here too?

**Review by cwperks — COMMENTED** [review]
Took a first pass. Thank you for adding missing test! This change looks good to me, just left some minor comments.

**Comment by prudhvigodithi** [comment]
Thanks @cwperks for reviewing the PR, initially I have to update the access modifiers to make sure those variables/methods are accessible for testing, I have reverted back and replaced to get the getters, can you review again?

File: src/test/java/org/opensearch/jobscheduler/JobSchedulerPluginTests.java (hunk 1 above)
Comment (prudhvigodithi) [reply src/test/java/org/opensearch/jobscheduler/JobSchedulerPluginTests.java:90]: Let me check this.

Comment (prudhvigodithi) [reply src/test/java/org/opensearch/jobscheduler/JobSchedulerPluginTests.java]: Just pushed in my latest commit.
Thanks

**Review by cwperks — APPROVED** [review]
